### PR TITLE
IGNITE-16689 Get rid of guava library in sql-engine module

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

File: modules/configuration-annotation-processor/src/integrationTest/java/org/apache/ignite/internal/configuration/processor/AbstractProcessorTest.java [hunk 1]
```diff
@@ -133,7 +132,7 @@ public BatchCompilation(List<ClassName> schemaClasses, Compilation compilation)
             generatedSources = compilation.generatedSourceFiles();
 
             generatedClasses = generatedSources.stream()
-                    .collect(Collectors.toMap(object -> fromGeneratedFilePath(object.getName()), Functions.identity()));
+                    .collect(Collectors.toMap(object -> fromGeneratedFilePath(object.getName()), o -> o));
```
Comment (yozaneaux) [inline modules/configuration-annotation-processor/src/integrationTest/java/org/apache/ignite/internal/configuration/processor/AbstractProcessorTest.java:135]: You could use `Function.identity()` from `import java.util.function.Function;` package.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItDataSchemaSyncTest.java [hunk 2]
```diff
@@ -132,7 +131,7 @@ void beforeEach() throws Exception {
      */
     @AfterEach
     void afterEach() throws Exception {
-        IgniteUtils.closeAll(Lists.reverse(clusterNodes));
+        IgniteUtils.closeAll(clusterNodes);
```
Comment (yozaneaux) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItDataSchemaSyncTest.java:134]: If you want to keep the reverse, you could use:
```
        Collections.reverse(clusterNodes);
        IgniteUtils.closeAll(clusterNodes);
```

**Review by yozaneaux — COMMENTED** [review]
Hi @zstan ,

You could use Function.identity() from jdk.

Regards

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItDataSchemaSyncTest.java (hunk 2 above)
Comment (zstan) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItDataSchemaSyncTest.java:134]: reverse - was a kind of hack from past ) Now it`s not an issue any more.

**Review by AMashenkov — APPROVED** [review]

**Pull request closed without merging**